src/components/CreateNewContactFormPage.jsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+function CreateNewContactFormPage() {
+  const URL = `https://boolean-api-server.fly.dev/llllllll-l/contact`;
+  const navigate = useNavigate();
+  const [formData, setFormData] = useState({
+    firstName: "",
+    lastName: "",
+    city: "",
+    street: "",
+    gender: "",
+    email: "",
+  });
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    if (formData.firstName === "") {
```

Comment:
Throwing errors directly in the event handler for missing fields will cause the application to crash if one of the conditions is met. Instead of throwing an error, consider displaying a user-friendly message. This could be done by introducing a new state to hold error messages:

`const [error, setError] = useState("");
`
`if (formData.firstName === "") {
  setError("First name cannot be empty");
  return; // Prevent form from submitting
} // Repeat for other fields
else {
  setError(""); // Clear any existing errors
}`